ts/sagas/startup/checkAcknowledgedEmailSaga.ts

```diff
@@ -0,0 +1,36 @@
+import { Effect } from "redux-saga";
+import { put, take } from "redux-saga/effects";
+
+import { UserProfileUnion } from "../../api/backend";
+import { navigateToEmailScreen } from "../../store/actions/navigation";
+import { emailAcknowledged } from "../../store/actions/onboarding";
+
+/**
+ * Launch email saga that consists of:
+ * - acknowledgement screen if email already exists and is valid and the user
+ *   does not want to change it
+ * - editing and validation screen, otherwise
+ */
+export function* checkAcknowledgedEmailSaga(
+  userProfile: UserProfileUnion
+): IterableIterator<Effect> {
+  // Check if user profile has email
+  if (userProfile.spid_email && userProfile.spid_email !== "") {
```

Comment:
Since spid_email is [EmailString](https://github.com/teamdigitale/io-ts-commons/blob/574a580fb95e49ad7be52d8990e9293396c5eb2d/src/strings.ts#L75) type you could avoid this check (the whole if statement), the codec ensures it for you.

here some tests I made in past
```typescript
describe("Email String", () => {
  it("should match a pattern", () => {
    const emptyEmail = "";
    const maybeEmail = EmailString.decode(emptyEmail);
    expect(maybeEmail.isLeft()).toBeTruthy();

    const emptyEmail2 = undefined;
    const maybeEmail2 = EmailString.decode(emptyEmail2);
    expect(maybeEmail2.isLeft()).toBeTruthy();

    const emptyEmail3 = "wrongEmail";
    const maybeEmail3 = EmailString.decode(emptyEmail3);
    expect(maybeEmail3.isLeft()).toBeTruthy();

    const gooEmail = "good@email.com";
    const maybeEmail5 = EmailString.decode(gooEmail);
    expect(maybeEmail5.isRight()).toBeTruthy();
  });
});
```

Reply:
so, the whole if statement would change to `maybeEmail.isRight()`, right? @Undermaken

Reply:
R: as discussed offline, I'm going to remove the statement in my next commit